consensus/common_test.go

```diff
@@ -681,6 +699,24 @@ func ensureVote(voteCh <-chan tmpubsub.Message, height int64, round int32,
 	}
 }
 
+func ensurePrevoteWithNilBlock(voteCh <-chan tmpubsub.Message) {
+	select {
+	case <-time.After(ensureTimeout):
+		panic("Timeout expired while waiting for NewVote event")
+	case msg := <-voteCh:
+		voteEvent, ok := msg.Data().(types.EventDataVote)
+		if !ok {
+			panic(fmt.Sprintf("expected a EventDataVote, got %T. Wrong subscription channel?",
+				msg.Data()))
+		}
+
+		vote := voteEvent.Vote
+		if len(vote.BlockID.Hash) == 0 {
```

Comment:
I think you should panic if hash if not nil, then at the callsite you don’t need to `assertPanic`